modules/setupadBidAdapter.js

```diff
@@ -0,0 +1,299 @@
+
+  getUserSyncs: function (syncOptions, responses, gdprConsent, uspConsent) {
+    if (!responses?.length) return [];
+
+    const syncs = [];
+    const bidders = getBidders(responses);
+
+    if (syncOptions.iframeEnabled && bidders) {
+      const queryParams = [];
+
+      queryParams.push(`bidders=${bidders}`);
+      queryParams.push('gdpr=' + +gdprConsent.gdprApplies);
+      queryParams.push('gdpr_consent=' + gdprConsent.consentString);
+      queryParams.push('usp_consent=' + (uspConsent || ''));
+
+      const strQueryParams = queryParams.join('&');
+
+      syncs.push({
+        type: 'iframe',
+        url: SYNC_ENDPOINT + strQueryParams + '&type=iframe',
+      });
+
+      return syncs;
+    }
+
+    return [];
+  },
+
+  onBidWon: function (bid) {
+    let bidder = bid.bidder || bid.bidderCode;
+    const auctionId = bid.auctionId;
+    if (bidder !== BIDDER_CODE) return;
+
+    let params;
+    if (bid.params) {
+      params = Array.isArray(bid.params) ? bid.params : [bid.params];
+    } else {
+      if (Array.isArray(bid.bids)) {
+        params = bid.bids.map((singleBid) => singleBid.params);
+      }
+    }
+
+    if (!params?.length) return;
+
+    const placementIdsArray = [];
+    params.forEach((param) => {
+      if (!param.placement_id) return;
+      placementIdsArray.push(param.placement_id);
+    });
+
+    const placementIds = (placementIdsArray.length && placementIdsArray.join(';')) || '';
+
+    if (!placementIds) return;
+
+    let extraBidParams = '';
+
+    // Iterate through all bidders to find the winning bidder by using creativeId as identification
+    for (const bidderName in biddersCreativeIds) {
+      if (
+        biddersCreativeIds.hasOwnProperty(bidderName) &&
+        biddersCreativeIds[bidderName] === bid.creativeId
+      ) {
+        bidder = bidderName;
+        break; // Exit the loop if a match is found
+      }
+    }
+
+    // Add extra parameters
+    extraBidParams = `&cpm=${bid.originalCpm}&currency=${bid.originalCurrency}`;
+
+    const url = `${REPORT_ENDPOINT}?event=bidWon&bidder=${bidder}&placementIds=${placementIds}&auctionId=${auctionId}${extraBidParams}&timestamp=${Date.now()}`;
+    triggerPixel(url);
+  },
+};
+
+function getBidders(serverResponse) {
+  const bidders = serverResponse
+    .map((res) => Object.keys(res.body.ext.responsetimemillis || []))
+    .flat(1);
+
+  if (bidders.length) {
+    return encodeURIComponent(JSON.stringify([...new Set(bidders)]));
+  }
+}
+
+function getAd(bid) {
+  let ad, adUrl, vastXml, vastUrl;
+
+  switch (deepAccess(bid, 'ext.prebid.type')) {
+    default:
+      if (bid.adm && bid.nurl) {
+        ad = bid.adm;
+        ad += createTrackPixelHtml(decodeURIComponent(bid.nurl));
+      } else if (bid.adm) {
+        ad = bid.adm;
+      } else if (bid.nurl) {
+        adUrl = bid.nurl;
+      }
+  }
+
+  return { ad, adUrl, vastXml, vastUrl };
+}
+
+function getSiteObj() {
+  const refInfo = (getRefererInfo && getRefererInfo()) || {};
```

Comment:
You probably do not need to import `getRefererInfo` as on the `bidderRequest` object there should be the referer info you are looking for.

Example:

![image](https://github.com/prebid/Prebid.js/assets/13972794/d178fa06-f503-42f7-9d09-517e9daa011d)


So maybe you can just pass `bidderRequest.refererInfo` into `getSiteObj` as `refInfo` in the func declaration and get rid of the import?